langchain/src/document.ts

```diff
@@ -17,12 +21,23 @@ export class Document<
 {
   pageContent: string;
 
+  id?: string;
```

Comment:
@dqbd to keep it simple I would keep one class. The fields are optionnal and the next step would be to update the document loaders to feed more info into this class

Reply:
I'm with David, let's keep this general and in line with Python's implementation:

https://github.com/hwchase17/langchain/blob/master/langchain/schema.py#L269

Is there a reason these extra fields can't just go into `metadata`?

Reply:
Either way, it's out of scope for this PR. I'm up for opening another one and having a more focused discussion later!

Reply:
I don't want to put everything into metadata for few reasons:
- It's a easier for dev to understand columns rather than property inside a JSON object
- it's easier to manipulate the data in columns than in a JSON
- it will be easier for dev to enforce indexes on columns 
- SQL visualization tools work out of the box for columns but some of them won't work as fine for Json object

I need it to add the concept of not recreating all embeddings.